packages/cli-core/src/package-manager-controller/package_manager_controller_factory.ts

```diff
@@ -57,7 +57,10 @@ export class PackageManagerControllerFactory {
   private getPackageManagerName() {
     const userAgent = process.env.npm_config_user_agent;
     if (userAgent === undefined) {
-      throw new Error('npm_config_user_agent is undefined');
+      throw new AmplifyUserError('NoPackageManagerError', {
+        resolution: `This is usually caused by attempting to run \`amplify\` directly. Try running \`npx amplify\``,
```

Comment:
nit: could use `format.command('npx amplify')`

Reply:
We have `format.runner('npx').amplifyCommand('sandbox')` for example, but that requires the caller to have resolved which package manager is being used. Which is what we are trying to get from the `npm_config_user_agent` but it's not set, so I'm not sure how we can do any better in this case.

Reply:
`Try running with package manager, for example \`npx amplify\`